src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Form.cs

```diff
@@ -0,0 +1,271 @@
+        /// Represents whether this is a required input
+        /// </summary>
+        public bool IsRequired;
+
+        /// <summary>
+        /// Represents the label of the input 
+        /// </summary>
+        public string Label;
+
+        /// <summary>
+        /// Combines the parent form id and input id
+        /// </summary>
+        public string CombinedId;
+
+        /// <summary>
+        /// Creates an input with given id and input type
+        /// </summary>
+        /// <param name="id">unique id for the input</param>
+        /// <param name="inputType">Input type for the input</param>
+        /// <param name="label">The label of the input</param>
+        /// <param name="isRequired">Indicates whether this is a required input</param>
+        public FormInput(int id, InputTypes inputType, string label = "", bool isRequired = false)
+        {
+            InputId = id;
+            InputType = inputType;
+            Label = label;
+            IsRequired = isRequired;
+        }
+    }
+
+    /// <summary>
+    /// Textbox class used to create a textbox input
+    /// </summary>
+    /// <example>
+    /// This sample shows how to create a textbox input
+    /// <code>
+    /// Textbox input1 = new Textbox(1, "Role Instance name", true);
+    /// </code>
+    /// </example>
+    public class Textbox: FormInput
+    {
+        /// <summary>
+        /// Creates an instance of Textbox class
+        /// </summary>
+        /// <param name="id">Unique id for the input</param>
+        /// <param name="label">Label for the textbox</param>
+        /// <param name="isRequired">Indicates if it is a required input</param>
+        public Textbox(int id, string label = "", bool isRequired = false): base(id, InputTypes.TextBox, label, isRequired)
+        {
+
+        }
+    }
+
+    /// <summary>
+    /// Button class used to create a button input
+    /// </summary>
+    /// <example>
+    /// This sample shows how to create a button input
+    /// <code>
+    /// Button saveButton = new Button(1, "Save");
+    /// </code>
+    /// </example>
+    public class Button: FormInput
+    {
+        /// <summary>
+        /// Sets the bootstrap button style
+        /// </summary>
+        public ButtonStyles ButtonStyle;
+
+        /// <summary>
+        /// Creates an instance of button class 
+        /// </summary>
+        /// <param name="id">Unique id for the button</param>
+        /// <param name="label">Label to display on the button</param>
+        /// <param name="buttonStyle">Bootstrap button style for the button</param>
+        public Button(int id, string label = "", ButtonStyles buttonStyle = ButtonStyles.Primary): base(id, InputTypes.Button, label, false)
+        {
+            this.ButtonStyle = buttonStyle;
+        }
+
+    }
+
+    public enum InputTypes
+    {
+        TextBox = 0,
+        Checkbox,
+        RadioButton,
+        DropDown,
+        Button
+    }
+
+    public enum ButtonStyles
```

Comment:
nitpick : Should we call this as StyleClass instead of ButtonStyles.
I see this getting used at multiple places and not only in just forms. So maybe taking this enum out into a separate file makes sense

Reply:
Shouldnt it be StyleClasses then? And also do you mean moving to a separate file under ResponseExtension namespace?

Reply:
Correct. 
You canuse them right now with button but the same set of bootstrap classes can be used with multiple other UI components that Bootstrap provides.